packages/ai-ide/src/browser/workspace-search-provider.ts

```diff
@@ -0,0 +1,132 @@
+// *****************************************************************************
+// Copyright (C) 2024 EclipseSource GmbH.
+//
+// This program and the accompanying materials are made available under the
+// terms of the Eclipse Public License v. 2.0 which is available at
+// http://www.eclipse.org/legal/epl-2.0.
+//
+// This Source Code may also be made available under the following Secondary
+// Licenses when the conditions for such availability set forth in the Eclipse
+// Public License v. 2.0 are satisfied: GNU General Public License, version 2
+// with the GNU Classpath Exception which is available at
+// https://www.gnu.org/software/classpath/license.html.
+//
+// SPDX-License-Identifier: EPL-2.0 OR GPL-2.0-only WITH Classpath-exception-2.0
+// *****************************************************************************
+
+import { ToolProvider, ToolRequest } from '@theia/ai-core';
+import { inject, injectable } from '@theia/core/shared/inversify';
+import { SearchInWorkspaceService, SearchInWorkspaceCallbacks } from '@theia/search-in-workspace/lib/browser/search-in-workspace-service';
+import { SearchInWorkspaceResult, SearchInWorkspaceOptions } from '@theia/search-in-workspace/lib/common/search-in-workspace-interface';
+
+const SEARCH_IN_WORKSPACE_ID = 'searchInWorkspace';
+@injectable()
+export class WorkspaceSearchProvider implements ToolProvider {
+
+    @inject(SearchInWorkspaceService)
+    protected readonly searchService: SearchInWorkspaceService;
+
+    private readonly MAX_RESULTS = 50;
+
+    getTool(): ToolRequest {
+        return {
+            id: SEARCH_IN_WORKSPACE_ID,
+            name: SEARCH_IN_WORKSPACE_ID,
+            description: 'Searches the workspace for files containing the provided query string or regular expression.',
```

Comment:
I had a few cases where the AI tried to search for multi-word search strings. So we should probably guide the AI here a bit better how to use this search tool. Maybe the following works better?

```suggestion
            description: 'Searches the workspace for files matching the given search term (parameter `query`), using case-insensitive full-string or regex matching. Multi-word patterns must match exactly (including spaces, but case-insensitive). For best results, prefer multiple queries with alternative search terms, over single long or complex ones.',
```

But I didn't deeply test this.